contents/rust/Rust.md

```diff
@@ -0,0 +1,80 @@
+# Rust
+
+Author(s): [Tan Li Hao](https://github.com/LiHaoTan)
+
+## Why Rust
+Rust is a relatively new programming language, so as such the benefits of learning or using the language is not very clear. It is also important to know the merits of a language so we know when to use them.
+
+The main merits would be in its [design](https://www.rust-lang.org/en-US/faq.html#what-is-this-projects-goal):
+* [safe](#safe) (e.g. memory safe)
+* [concurrent](#concurrent)
+* [practical](#practical)
+
+### Safe
+
+Designing the language to be safe allows common errors such as segmentation faults ([[1]](https://en.wikipedia.org/wiki/Segmentation_fault), [[2]](https://stackoverflow.com/questions/2346806/what-is-a-segmentation-fault/2346849#2346849)), [resource leaks](http://blog.skylight.io/rust-means-never-having-to-close-a-socket/), and [many others](https://www.reddit.com/r/rust/comments/2mwpie/what_are_the_advantages_of_rust_over_modern_c/) to be avoided. Many of these pitfalls can already be avoided in modern languages such as [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) in C++ and [try-with-resources](https://blogs.oracle.com/darcy/more-concise-try-with-resources-statements-in-jdk-9) in Java, but mostly still requires developer discipline to ensure safety ([[1]](https://www.rust-lang.org/en-US/faq.html#why-rust-vs-cxx), [[2]](https://www.reddit.com/r/rust/comments/2mwpie/what_are_the_advantages_of_rust_over_modern_c/)).
+
+The underlying concept for the safety provided in Rust is called [ownership](https://doc.rust-lang.org/book/second-edition/ch04-00-understanding-ownership.html) and is [translatable to other languages](https://codewithoutrules.com/2017/01/26/object-ownership/), and safety is ensured by Rust compile-time.
+
+### Concurrent
+
+Concurrency is getting [increasingly important](https://softwareengineering.stackexchange.com/questions/115474/why-should-i-know-concurrent-programming) but it is challenging to write concurrent code ([[1]](https://news.ycombinator.com/item?id=8138578), [[2]](https://golang.org/doc/faq#csp), [[3]](http://joeduffyblog.com/2016/11/30/15-years-of-concurrency/)).
```

Comment:
Would it be good to mention the (memory) safety concepts and the challenge briefly? This can keep this document more self-contained. If not it will mean that the reader has to enter the links to read this document through smoothly.

Reply:
I tried to address this by mentioning where they generally fall short and the idea behind the safety concept, not sure if this makes it better?